qutip/tests/test_lattice.py

```diff
@@ -0,0 +1,713 @@
+#    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+import numpy as np
+from qutip.lattice import *
+from qutip import (Qobj, tensor, basis, qeye, isherm)
+import random
+#from numpy.testing import (assert_equal, assert_, assert_almost_equal,
+#                            run_module_suite)
+
+from numpy.testing import (assert_, run_module_suite)
+
+class TestLattice:
+    """
+    Tests for `qutip.lattice` class.
+    """
+    def test_hamiltonian(self):
+        """
+        lattice: Test the method Lattice1d.Hamiltonian().
+        """
+        # num_cell = 1
+        # Four different instances
+        Periodic_Atom_Chain = Lattice1d(num_cell=1, boundary = "periodic")
+        Aperiodic_Atom_Chain = Lattice1d(num_cell=1, boundary = "aperiodic")        
+        p_1223 = Lattice1d(num_cell=1, boundary = "periodic",
+                                 cell_num_site = 2, cell_site_dof = [2,3])
+        ap_1223 = Lattice1d(num_cell=1, boundary = "aperiodic",
+                                 cell_num_site = 2, cell_site_dof = [2,3])
+
+        # Their Hamiltonians
+        pHamt1 = Periodic_Atom_Chain.Hamiltonian()
+        apHamt1 = Aperiodic_Atom_Chain.Hamiltonian()
+        pHamt1223 = p_1223.Hamiltonian()
+        apHamt1223 = ap_1223.Hamiltonian()
+
+        # Benchmark answers
+        pHamt1_C = Qobj([[0.]],dims=[[1], [1]])
+        apHamt1_C = Qobj([[0.]],dims=[[1], [1]])
+        site1223 = np.diag(np.zeros(2-1)-1, 1) + np.diag(np.zeros(2-1)-1, -1)
+        Rpap1223 = tensor(Qobj(site1223), qeye([2, 3]))
+        pap1223 = Qobj(Rpap1223, dims=[[2, 2, 3], [2, 2, 3]])
+        
+        # Checks for num_cell = 1
+        assert_(pHamt1 == pHamt1_C)
+        assert_(apHamt1 == apHamt1_C)
+        assert_(pHamt1223 == pap1223)    # for num_cell=1, periodic and
+        assert_(apHamt1223 == pap1223)   # aperiodic B.C. have same Hamiltonian
+        
+        # num_cell = 2
+        # Four different instances
+        Periodic_Atom_Chain = Lattice1d(num_cell=2, boundary = "periodic")
+        Aperiodic_Atom_Chain = Lattice1d(num_cell=2, boundary = "aperiodic")        
+
+        p_2222 = Lattice1d(num_cell=2, boundary = "periodic",
+                                 cell_num_site = 2, cell_site_dof = [2,2])
+        ap_2222 = Lattice1d(num_cell=2, boundary = "aperiodic",
+                                 cell_num_site = 2, cell_site_dof = [2,2])
+
+        # Their Hamiltonians
+        pHamt2222 = p_2222.Hamiltonian()
+        apHamt2222 = ap_2222.Hamiltonian()
+        pHamt2 = Periodic_Atom_Chain.Hamiltonian()
+        apHamt2 = Aperiodic_Atom_Chain.Hamiltonian()
+        
+        # Benchmark answers        
+        pHamt2_C = Qobj([[ 0., -1.],
+        [-1.,  0.]],dims = [[2], [2]])
+        apHamt2_C = Qobj([[ 0., -1.],
+        [-1.,  0.]],dims = [[2], [2]])
+
+        pap_2222 = Qobj(np.array([[
```

Comment:
For big near empty matrix:
```
mat = np.zeros((16,16), dtype=complex)
mat[0,4] = -1
...
```
Take less place and easier to read.

Reply:
Updated. Thanks.